features/support/twopence_init.rb

```diff
@@ -1,50 +1,57 @@
 require 'twopence'
 
-# initialize ssh targets from environment variables.
-
-raise 'Server ip var empty' if ENV['TESTHOST'].nil?
-raise 'client ip var empty' if ENV['CLIENT'].nil?
-raise 'minion ip var empty' if ENV['MINION'].nil?
-raise 'ceos-minion ip var empty' if ENV['CENTOSMINION'].nil?
-raise 'ssh-minion ip var empty' if ENV['SSHMINION'].nil?
+# Initialize SSH targets from environment variables
+# If there is no proxy, use "None" for $PROXY environment variable
+raise 'Server IP address or domain name variable empty' if ENV['TESTHOST'].nil?
+raise 'Proxy IP address or domain name variable empty' if ENV['PROXY'].nil?
+raise 'Client IP address or domain name variable empty' if ENV['CLIENT'].nil?
+raise 'Minion IP address or domain name variable empty' if ENV['MINION'].nil?
+raise 'CentOS minion IP address or domain name variable empty' if ENV['CENTOSMINION'].nil?
+raise 'SSH minion IP address or domain name variable empty' if ENV['SSHMINION'].nil?
 
 $server_ip = ENV['TESTHOST']
+$proxy_ip = ENV['PROXY']
 $client_ip = ENV['CLIENT']
 $minion_ip = ENV['MINION']
 $ceos_minion_ip = ENV['CENTOSMINION']
 $ssh_minion_ip = ENV['SSHMINION']
 
-# define twopence object.
+# Define twopence objects
 $client = Twopence.init("ssh:#{$client_ip}")
+$proxy = Twopence.init("ssh:#{$proxy_ip}") if $proxy_ip != 'None'
```

Comment:
``` unless proxy_ip

Reply:
no, it is "None" - dictated by Salt

Reply:
@Bischoff can you point me where in salt you get this Grains variable?

Reply:
sumaform/salt/controller/bashrc
```
export PROXY={{ grains.get("proxy") }}
```

"None" is not harcoded here - it's what Salt gives when the grain is missing.

Reply:
There is a way to  set defaults in saltstack when the grains is None. 
atm cannot remember